llvm/test/lab3/kazantsev_evgeny/Mtr.mir

```diff
@@ -0,0 +1,381 @@
+callSites:       []
+debugValueSubstitutions: []
+constants:       []
+machineFunctionInfo: {}
+body:             |
+  bb.0.entry:
+    liveins: $xmm0, $xmm1, $xmm3
+  
+    ; CHECK:        %3:vr128 = COPY $xmm3
+    ; CHECK-NOT:    VFMADD213PDr
+    ; CHECK:        RET 0, $xmm0
+    %3:vr128 = COPY $xmm3
+    %1:vr128 = COPY $xmm1
+    %0:vr128 = COPY $xmm0
+    %4:vr128 = nofpexcept MULPDrr %0, %1, implicit $mxcsr
+    %5:vr128 = nofpexcept MULPDrr %1, %3, implicit $mxcsr
+    %6:vr128 = nofpexcept ADDPDrr %4, %0, implicit $mxcsr
+    %7:vr128 = nofpexcept SUBPDrr %5, killed %6, implicit $mxcsr
+    %8:vr128 = nofpexcept ADDPDrr %4, killed %7, implicit $mxcsr
+    $xmm0 = COPY %8
+    RET 0, $xmm0
+
+...
+---
+name:            roar
+alignment:       16
+exposesReturnsTwice: false
+legalized:       false
+regBankSelected: false
+selected:        false
+failedISel:      false
+tracksRegLiveness: true
+hasWinCFI:       false
+callsEHReturn:   false
+callsUnwindInit: false
+hasEHCatchret:   false
+hasEHScopes:     false
+hasEHFunclets:   false
+isOutlined:      false
+debugInstrRef:   true
+failsVerification: false
+tracksDebugUserValues: false
+registers:
+  - { id: 0, class: vr128, preferred-register: '' }
+  - { id: 1, class: vr128, preferred-register: '' }
+  - { id: 2, class: vr128, preferred-register: '' }
+  - { id: 3, class: vr128, preferred-register: '' }
+  - { id: 4, class: vr128, preferred-register: '' }
+  - { id: 5, class: vr128, preferred-register: '' }
+  - { id: 6, class: vr128, preferred-register: '' }
+  - { id: 7, class: vr128, preferred-register: '' }
+liveins:
+  - { reg: '$xmm0', virtual-reg: '%0' }
+  - { reg: '$xmm1', virtual-reg: '%1' }
+  - { reg: '$xmm3', virtual-reg: '%3' }
+frameInfo:
+  isFrameAddressTaken: false
+  isReturnAddressTaken: false
+  hasStackMap:     false
+  hasPatchPoint:   false
+  stackSize:       0
+  offsetAdjustment: 0
+  maxAlignment:    1
+  adjustsStack:    false
+  hasCalls:        false
+  stackProtector:  ''
+  functionContext: ''
+  maxCallFrameSize: 4294967295
+  cvBytesOfCalleeSavedRegisters: 0
+  hasOpaqueSPAdjustment: false
+  hasVAStart:      false
+  hasMustTailInVarArgFunc: false
+  hasTailCall:     false
+  localFrameSize:  0
+  savePoint:       ''
+  restorePoint:    ''
+fixedStack:      []
+stack:           []
+entry_values:    []
+callSites:       []
+debugValueSubstitutions: []
+constants:       []
+machineFunctionInfo: {}
+body:             |
+  bb.0.entry:
+    liveins: $xmm0, $xmm1, $xmm3
+  
+    %3:vr128 = COPY $xmm3
+    %1:vr128 = COPY $xmm1
+    %0:vr128 = COPY $xmm0
+    ; CHECK:        %0:vr128 = COPY $xmm0
+    ; CHECK-NOT:    %4:vr128 = nofpexcept MULPDrr %1, %3, implicit $mxcsr
+    %4:vr128 = nofpexcept MULPDrr %1, %3, implicit $mxcsr
+    ; CHECK:        %5:vr128 = nofpexcept ADDPDrr %0, %3, implicit $mxcsr
+    ; CHECK-NEXT:   %6:vr128 = VFMADD213PDr %1, %3, %5, implicit $mxcsr
+    ; CHECK-NEXT:   %7:vr128 = nofpexcept MULPDrr %5, killed %6, implicit $mxcsr
```

Comment:
What if instruction will be the following:
```suggestion
    ; CHECK-NEXT:   %7:vr128 = nofpexcept MULPDrr %5, killed %4, implicit $mxcsr
```

Reply:
The test assumes that the VFMADD213PDr instruction (previous line) will be followed by a MULPDrr instruction and that the register specified as %6 will be destroyed or corrupted in this instruction. However, the kill attribute actually points to register %4, which is not the expected string in the test.

Reply:
Ok, nevermind.